compaction/compaction.cc

```diff
@@ -1643,7 +1643,7 @@ future<bool> validate_compaction_validate_reader(flat_mutation_reader reader, co
         while (auto mf_opt = co_await reader(db::no_timeout)) {
             if (info.is_stop_requested()) [[unlikely]] {
                 // Compaction manager will catch this exception and re-schedule the compaction.
-                throw compaction_stop_exception(info.ks_name, info.cf_name, info.stop_requested);
+                co_return coroutine::exception(std::make_exception_ptr(compaction_stop_exception(info.ks_name, info.cf_name, info.stop_requested)));
```

Comment:
A thought: we can have a shortcut coroutine::make_exception() which combines coroutine::exception and std::make_exception_ptr(), and makes this idiom a bit shorter.

Reply:
Indeed, when writing this series I wished I had also added a `coroutine::make_exception` template. It's a good idea, so I'll proceed with sending a Seastar patch and then, once it's in, I'll rewrite all these idioms.

Reply:
Sure, it doesn't need to be now.

Reply:
Ok, let me verify what happens.